libsignal-service/src/groups_v2/model.rs

```diff
@@ -0,0 +1,246 @@
+            Self::NewRequestingMember(arg0) => {
+                f.debug_tuple("NewRequestingMember").field(arg0).finish()
+            },
+            Self::DeleteRequestingMember(arg0) => {
+                f.debug_tuple("DeleteRequestingMember").field(arg0).finish()
+            },
+            Self::PromoteRequestingMember { uuid, role } => f
+                .debug_struct("PromoteRequestingMember")
+                .field("uuid", uuid)
+                .field("role", role)
+                .finish(),
+            Self::Title(arg0) => f.debug_tuple("Title").field(arg0).finish(),
+            Self::Avatar(arg0) => f.debug_tuple("Avatar").field(arg0).finish(),
+            Self::Timer(arg0) => f.debug_tuple("Timer").field(arg0).finish(),
+            Self::Description(arg0) => {
+                f.debug_tuple("Description").field(arg0).finish()
+            },
+            Self::AttributeAccess(arg0) => {
+                f.debug_tuple("AttributeAccess").field(arg0).finish()
+            },
+            Self::MemberAccess(arg0) => {
+                f.debug_tuple("MemberAccess").field(arg0).finish()
+            },
+            Self::InviteLinkAccess(arg0) => {
+                f.debug_tuple("InviteLinkAccess").field(arg0).finish()
+            },
+            Self::InviteLinkPassword(arg0) => {
+                f.debug_tuple("InviteLinkPassword").field(arg0).finish()
+            },
+            Self::AnnouncementOnly(arg0) => {
+                f.debug_tuple("AnnouncementOnly").field(arg0).finish()
+            },
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct Timer {
+    pub duration: u32,
+}
+
+/// Conversion from protobuf definitions
+
+impl TryFrom<i32> for Role {
+    type Error = GroupDecryptionError;
+
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        use crate::proto::member::Role::*;
+        match crate::proto::member::Role::from_i32(value) {
+            Some(Unknown) => Ok(Role::Unknown),
+            Some(Default) => Ok(Role::Default),
+            Some(Administrator) => Ok(Role::Administrator),
+            None => Err(GroupDecryptionError::WrongEnumValue),
+        }
+    }
+}
+
+impl TryFrom<i32> for AccessRequired {
+    type Error = GroupDecryptionError;
+
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        use crate::proto::access_control::AccessRequired::*;
+        match crate::proto::access_control::AccessRequired::from_i32(value) {
+            Some(Unknown) => Ok(AccessRequired::Unknown),
+            Some(Any) => Ok(AccessRequired::Any),
+            Some(Member) => Ok(AccessRequired::Member),
+            Some(Administrator) => Ok(AccessRequired::Administrator),
+            Some(Unsatisfiable) => Ok(AccessRequired::Unsatisfiable),
+            None => Err(GroupDecryptionError::WrongEnumValue),
+        }
+    }
+}
+
+impl TryFrom<crate::proto::AccessControl> for AccessControl {
+    type Error = GroupDecryptionError;
```

Comment:
I feel like this might have to be a `ServiceError` instead?

Reply:
Done!